src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/event/TimelineEventResolver.kt

```diff
@@ -0,0 +1,109 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.event
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.GoalStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Step
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.StepStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEventType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEventType.GOAL_UPDATED
+import java.util.UUID
+
+/**
+ * Responsible for identifying which [TimelineEvent]s have occurred, for example following an update to a [Goal].
+ */
+@Component
+class TimelineEventResolver {
```

Comment:
Oh, so this resolver is only responsible for working out and resolving **update** type events? Is that right?
I was wondering why the other methods that create TimelineEvents (that I've previously commented might suit a mapper?) were not in here .... 🤔